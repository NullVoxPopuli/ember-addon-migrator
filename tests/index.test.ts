import fse from 'fs-extra';
import { beforeAll,describe, expect, test } from 'vitest';

import { type Project, addonFrom, build,emberTest, findFixtures, lintAddon, lintTestApp, migrate } from './helpers.js';

let fixtures = await findFixtures();

describe('fixtures', () => {
  for (let fixtureName of fixtures) {
    describe(fixtureName, () => {
      let project: Project;

      beforeAll(async () => {
        project = await addonFrom(fixtureName);
      });

      test('verify tmp project', async () => {
        expect(await fse.pathExists(project.rootPath), 'rootPath').toBe(true); 
        expect(await fse.pathExists(project.addonPath), 'addonPath').toBe(true); 
        expect(await fse.pathExists(project.testAppPath), 'testAppPath').toBe(true); 
      });

      test('migrate & build', async () => {
        await migrate(project);
        await build(project);

        // Yay! no errors!
        expect(true).toBe(true);
      });

      test.concurrent('lint addon', async () => {
        let result = await lintAddon(project)

        expect(result).toMatchObject({ exitCode: 0 });
      });

      test.concurrent('lint test-app', async () => {
        let result = await lintTestApp(project);

        expect(result).toMatchObject({ exitCode: 0 });
      });

      test.concurrent('tests pass', async () => {
        let result = await emberTest(project);

        expect(result).toMatchObject({ exitCode: 0 });
      });
    }) 
  }
})

