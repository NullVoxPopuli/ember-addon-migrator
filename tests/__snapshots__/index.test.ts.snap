// Vitest Snapshot v1

exports[`basic conversion > js addon 1`] = `
"[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
[5/5] Building fresh packages...
success Saved lockfile.
ðŸŽ‰ Congratulations! Your addon is now formatted as a V2 addon!

Next steps:
 - customize the rollup.config.js to suit your needs
 - audit dependencies of both the new addon and test-app
 - run \`yarn install\`
 - run \`yarn run lint:fix\` in each workspace
 - test your addon \`cd test-app; yarn test\`
 - push up a branch, merge, release
 - party ðŸ¥³"
`;
